bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerCreateOp.java

```diff
@@ -200,4 +209,220 @@ private void createComplete(int rc, LedgerHandle lh) {
         cb.createComplete(rc, lh, ctx);
     }
 
+    private static final byte[] EMPTY_PASSWORD = new byte[0];
+
+    public static class CreateBuilderImpl implements CreateBuilder {
+
+        private final BookKeeper bk;
+        private int builderEnsembleSize = 3;
+        private int builderAckQuorumSize = 2;
+        private int builderWriteQuorumSize = 2;
+        private byte[] builderPassword = EMPTY_PASSWORD;
```

Comment:
in the current API, password and DigestType are mandatory fields, not sure why they were mandatory parameters and is it ok to make it non-mandatory in the new API? 

@sijie

Reply:
password and digesttype are already on metadata. password are we already discussed is only for preventing accidental access to data it is not a security feature.
In the new API we can make them optional (that is with good defaults)

Reply:
@reddycharan you can see this as mandatory with defaults, no? builder should validate those fields before building the actual op.

Reply:
make the passwd `null`. and validate on builder